src/app/functions/server/featureFlags.ts

```diff
@@ -1,75 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-import {
-  FeatureFlag,
-  getFeatureFlagByName,
-} from "../../../db/tables/featureFlags";
-import { Session } from "next-auth";
-
-export type FeatureFlagsEnabled = {
-  FreeBrokerScan: boolean;
-  PremiumBrokerRemoval: boolean;
-  FalseDoorTest: boolean;
-  HibpBreachNotifications: boolean;
-};
-
-// This function is specifically meant to not execute in tests:
-/* c8 ignore next 6 */
-const warn: typeof console.warn = (...args) => {
-  if (process.env.NODE_ENV === "test") {
-    return;
-  }
-  console.warn(...args);
-};
-
-export async function isFlagEnabled(
-  name: keyof FeatureFlagsEnabled,
-  user?: Session["user"]
-): Promise<boolean> {
-  const data = await getFeatureFlagByName(name);
-
-  // TODO: Add unit test when changing this code:
-  /* c8 ignore next 4 */
-  if (!data) {
-    warn("Feature flag does not exist:", name);
-    return false;
-  }
-
-  const flag: FeatureFlag = {
-    name: data.name,
-    isEnabled: data.is_enabled,
-    description: data.description,
-    dependencies: data.dependencies,
-    allowList: data.allow_list,
-    waitList: data.wait_list,
-    expiredAt: data.expired_at,
-    deletedAt: data.deleted_at,
-    owner: data.owner,
-  };
-
-  if (flag.deletedAt) {
-    warn("Flag has been deleted:", flag.name);
-    return false;
-  }
-
-  if (flag.expiredAt) {
-    warn("Flag has expired:", flag.name);
-    return false;
-  }
-
-  if (!flag.isEnabled) {
-    warn("Flag is not enabled:", flag.name);
-    return false;
-  }
-
-  if (!user) {
-    return true;
```

Comment:
Just a note: as I understand it, this resulted in `isFlagEnabled` being `true` even if there was an `allow_list` for the flag, as long as you did not pass in a user. In my implementation (in `/db/tables/featureFlags.ts`), the equivalent of not passing in a user is to call `getFlagsEnabledForEveryone`, but it will return `false` if `allow_list` is a non-empty array. Is assume that that's the actual intended behaviour of the allowlist.

Reply:
That was what I was thinking originally, but I think in practice it ended up being based on the way `isFlagEnabled` is called - if the `session` argument is passed then the allow list is considered, and if it is not then the allow list is not.

I'd like to figure out how to be more explicit. Maybe we should just have a boolean column that signifies this is globally enabled or not?

Reply:
That sounds really unintuitive to me - that means you can add email addresses to an allowlist, and then accounts with different email addresses could still have features behind that flag show up.

But I updated the PR to align with that older behaviour, hopefully calling that out somewhat explicitly in the API by calling it `ignoreAllowlist`: 6b7315cb6

But yes, an explicit boolean that enables it for everyone (and clears the allowlist) would work too. Python's Waffle e.g. has an `--everyone` switch in addition to allowing enabling flags for specific users (or e.g. all authenticated users).